src/Recoil__React.re

```diff
@@ -16,10 +16,8 @@ external useRecoilState:
   Recoil__Value.readWrite('value) => ('value, ('value => 'value) => unit) =
   "useRecoilState";
 
-type value('a) = 'a;
```

Comment:
this is just drive-by cleanup, which also removes this extraneous type from the public API.

Reply:
I'm wondering if I didn't use that on purpose so that it forces BuckleScript not to curry if the value is a function, but I'm not sure anymore 😄.

Reply:
Looking at the code generated by tests it doesn't seem like that's the case. See e.g. `atom4`, just generates the following JS:

```js
  var atom4$1 = Recoil.useRecoilValue(atom4);
```

Reply:
Alright, looks ok!